contrib/hashicorp/vault/vault.go

```diff
@@ -0,0 +1,98 @@
+// The vault package contains functions to construct an *http.Client that will
+// integrate with the github.com/hashicorp/vault/api and collect traces to send
+// to DataDog.
+//
+// The easiest way to use this package is to create an *http.Client with
+// NewHttpClient, and put it in the vault api config that's passed to the
+// NewClient function.
+//    import "github.com/hashicorp/vault/api"
+//    import trace "gopkg.in/DataDog/dd-trace-go.v1/contrib/hashicorp/vault"
+//
+//    func ... {
+//        api.NewClient(&api.Config{HttpClient: trace.NewHttpClient()})
+//    }
+//
+// If you are already using your own *http.Client with vault, you can use the
+// WrapHttpTransport function to wrap the client with the tracer code. Your
+// *http.Client will continue to work as before, but will also capture traces.
+//
+//    func ... {
+//        var myHttpClient *http.Client
+//        myHttpClient = ...
+//        api.NewClient(&api.Config{HttpClient: trace.WrapHttpTransport(myHttpClient)})
+//    }
+//
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+	vault "github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+)
+
+// NewHttpClient returns an *http.Client for use in the config for the vault api
+// Client.
+func NewHttpClient() *http.Client {
+	dc := vault.DefaultConfig()
+	c := dc.HttpClient
+	WrapHttpTransport(c)
+	return c
+}
+
+type vaultErrors struct {
+	Errors []string `json:"errors"`
+}
+
+// WrapHttpTransport takes an existing *http.Client and wraps the transport with
+// the tracing code.
+func WrapHttpTransport(client *http.Client) *http.Client {
+	wrapped := httptrace.WrapRoundTripper(client.Transport,
+		httptrace.WithBefore(func(req *http.Request, span ddtrace.Span) {
+			span.SetTag(ext.ServiceName, "vault")
+			span.SetTag(ext.HTTPURL, req.URL.Path)
```

Comment:
Does the path every contain variable components? I think no, but just making sure.